### PR TITLE
refactor: handle repos restructure

### PR DESCRIPTION
**Pull request by blaisedias**
Refactor to facilitate creation of pipelines
capable of running system test on
 - mayastor repos
 - bolt repos

1 - parameterise selection of product in the generic-system-test pipeline
2 - add support function to set repository urls and paths
    on a product basis
        - mayastor
        - bolt
3 - refactor to allow use of mayastor-e2e branches
    - side effect - artifacts are now generated
        under mayastor-e2e/artifacts

Note: untested with datacore bolt repos.
Further changes may be required when the bolt repos are
usable for CI/CD

### CODE REVIEW

File: pipelines/common2/common.groovy [hunk 1]
```diff
@@ -10,6 +10,33 @@ def unwrap(Map params, key) {
...
+def GetProductSettings (datacore_bolt) {
+    if (datacore_bolt == true) {
+        return [
+            dataplane_dir: "bolt-data-plane",
+            dataplane_repo_url: "https://github.com/datacoresoftware/bolt-data-plane-test",
+            controlplane_dir: "bolt-control-plane",
+            controlplane_repo_url: "https://github.com/datacoresoftware/bolt-control-plane-test",
+        ]
+    }
+    return [
```
Comment (chriswldenyer) [inline pipelines/common2/common.groovy:22]: nit: I would be tempted to put an "else" here for symmetry

**Review by rohan2794 — APPROVED** [review]
LGTM

File: pipelines/common2/common.groovy [hunk 2]
```diff
@@ -574,6 +618,15 @@ def ParallelArchiveArtefacts(Map params) {
     def artefacts_stash_queue = unwrap(params,'artefacts_stash_queue')
     def passed_tests_queue = unwrap(params,'passed_tests_queue')
     def failed_tests_queue = unwrap(params,'failed_tests_queue')
+    def e2e_artifacts_dir = unwrap(params,'e2e_artifacts_dir')
+    def artifacts_dir ='artifacts'
```
Comment (chriswldenyer) [inline pipelines/common2/common.groovy:622]: Not sure that artifacts_dir needs to be a variable. It can easily be confused with e2e_artifacts_dir.

**Review by chriswldenyer — APPROVED** [review]
Looks OK, a couple of comments.

**Pull request merged**